### DISCUSSION TITLE
Incorrect position of the path parameters with a dependency

### DISCUSSION THREAD

**Discussion by ghost** [fastapi/fastapi#9767] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated
from fastapi import Depends, FastAPI, Path, HTTPException

app = FastAPI()

fake_devices_db = [
    {"device_id": 1, "items": [{"item_name": "Foo"}, {"item_name": "Bar"}, {"item_name": "Baz"}]},
    {"device_id": 2, "items": [{"item_name": "Foo"}, {"item_name": "Bar"}, {"item_name": "Baz"}]},
    {"device_id": 3, "items": [{"item_name": "Foo"}, {"item_name": "Bar"}, {"item_name": "Baz"}]},
]


def get_device(device_id: Annotated[int, Path(...)]):
    if devices := [d for d in fake_devices_db if d.get("device_id") == device_id]:
        return devices[0]
    raise HTTPException(status_code=404, detail="Device not found")


@app.get("/devices/{device_id}/items/{item_name}/")
async def read_items(device: Annotated[dict, Depends(get_device)], item_name: str):
    return {"items": [i for i in device.get("items") if i.get("item_name") == item_name]}
```

### Description

I would like that the position of the Path parameters was always device_id and item_id, as defined in the endpoint, but if the device_id is a dependency then the Swagger documentation generated by Fast API modifies the position of them.

Here you can see the openapi.json that generated the code of the example:

`{"openapi":"3.0.2","info":{"title":"FastAPI","version":"0.1.0"},"paths":{"/devices/{device_id}/items/{item_name}/":{"get":{"summary":"Read Items","operationId":"read_items_devices__device_id__items__item_name___get","parameters":[{"required":true,"schema":{"title":"Item Name","type":"string"},"name":"item_name","in":"path"},{"required":true,"schema":{"title":"Device Id","type":"integer"},"name":"device_id","in":"path"}],"responses":{"200":{"description":"Successful Response","content":{"application/json":{"schema":{}}}},"422":{"description":"Validation Error","content":{"application/json":{"schema":{"$ref":"#/components/schemas/HTTPValidationError"}}}}}}}},"components":{"schemas":{"HTTPValidationError":{"title":"HTTPValidationError","type":"object","properties":{"detail":{"title":"Detail","type":"array","items":{"$ref":"#/components/schemas/ValidationError"}}}},"ValidationError":{"title":"ValidationError","required":["loc","msg","type"],"type":"object","properties":{"loc":{"title":"Location","type":"array","items":{"anyOf":[{"type":"string"},{"type":"integer"}]}},"msg":{"title":"Message","type":"string"},"type":{"title":"Error Type","type":"string"}}}}}}`

### Operating System

Linux, macOS

### FastAPI Version

0.97.0

### Python Version

3.10.5

**Comment by abhint** [comment]
```py
@app.get("/devices/{device_id}/items/{item_name}/")
async def read_items(device_id: int = Path(...), item_name: str = Path(...)):
    device = get_device(device_id)
    return {"items": [i for i in device.get("items") if i.get("item_name") == item_name]}

 ```

**Reply by ghost** [reply]
Thanks for the response, I already know it, but I want to use the dependency. Also, I can add the device_id in the route method to force the order the following way...

```python
@app.get("/devices/{device_id}/items/{item_name}/")
async def read_items(device_id: int = Path(...), item_name: str = Path(...), device: Annotated[dict, Depends(get_device)]):
    return {"items": [i for i in device.get("items") if i.get("item_name") == item_name]}
``` 

....but I'm asking if is possible doesn't do it.

**Comment by YuriiMotov** [comment, marked as answer]
There is no simple way to specify the order of parameters in openapi schema.
One way you already discussed is to [add all these parameters to the endpoint function](https://github.com/fastapi/fastapi/discussions/9767#discussioncomment-6322618). Another way is to [modify openapi schema](https://github.com/fastapi/fastapi/discussions/13206#discussioncomment-11847220)
